projects/views/common.py

```diff
@@ -251,3 +258,47 @@ def project_unsubscribe_api_view(request: Request, pk: str):
     unsubscribe_project(project=project, user=request.user)
 
     return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@api_view(["GET"])
+@permission_classes([IsAuthenticated])
+def download_data(request, project_id: int):
+    user = request.user
+    qs = get_projects_qs(user=user)
+    obj = get_object_or_404(qs, pk=project_id)
+    # Check permissions
+    if not (
+        user.is_staff
+        or WhitelistUser.objects.filter(
+            Q(project=obj) | (Q(post__isnull=True) & Q(project__isnull=True)),
+            user=user,
+        ).exists()
+    ):
+        raise PermissionDenied("You are not allowed to download this project")
+
+    serializer = DownloadDataSerializer(data=request.GET)
```

Comment:
A small nit: let's use `request.query_params`:
> For clarity inside your code, we recommend using request.query_params instead of the Django's standard request.GET

https://www.django-rest-framework.org/api-guide/requests/#query_params